src/backend/src/lib.rs

```diff
@@ -580,14 +581,14 @@ fn get_user_profile() -> Result<UserProfile, GetUserProfileError> {
 #[query(guard = "caller_is_allowed")]
 #[allow(clippy::needless_pass_by_value)]
 fn get_users(request: GetUsersRequest) -> GetUsersResponse {
-    // TODO: Implement https://dfinity.atlassian.net/browse/GIX-2650
     // WARNING: The value `DEFAULT_LIMIT_GET_USERS_RESPONSE` must also be determined by the cycles consumption when reading BTreeMap.
-    let users: Vec<OisyUser> = Vec::new();
+
+    let (users, limit_users_size): (Vec<OisyUser>, u64) =
```

Comment:
Just ideas:

- You can rename `limit_users_size` to `matches_max_length` if interesting.

- Should we create a type `type OisyUsers = Vec<OisyUser>`?